src/bindings/python/src/openvino/opset16/ops.py

```diff
@@ -32,3 +32,33 @@ def identity(
         as_nodes(data, name=name),
         {},
     )
+
+
+@nameable_op
+def segment_max(
+    data: NodeInput,
+    segment_ids: NodeInput,
+    num_segments: Optional[NodeInput] = None,
+    fill_mode: Optional[str] = None,
```

Comment:
Seem like fill_mode is reqired, so maybe instead defaulting to None that raises error, it should default to one of possible values?

Also since this val could be either ZERO or LOWEST, consider using Literal, for example
```suggestion
    fill_mode: Optional[Literal["ZERO", "LOWEST"]] = None,
```
\+ required import for typing.Literal.